loan.java

```diff
@@ -4,75 +4,76 @@
-		this.ID = loanId;
-		this.B = book;
-		this.M = member;
-		this.D = dueDate;
-		this.state = LOAN_STATE.CURRENT;
-	}
-
-	
-	public void checkOverDue() {
-		if (state == LOAN_STATE.CURRENT &&
-			Calendar.getInstance().getDate().after(D)) {
-			this.state = LOAN_STATE.OVER_DUE;			
-		}
-	}
-
-	
-	public boolean OVer_Due() {
-		return state == LOAN_STATE.OVER_DUE;
-	}
-
-	
-	public Integer ID() {
-		return ID;
-	}
-
-
-	public Date Get_Due_Date() {
-		return D;
-	}
-	
-	
-	public String toString() {
-		SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
-
-		StringBuilder sb = new StringBuilder();
-		sb.append("Loan:  ").append(ID).append("\n")
-		  .append("  Borrower ").append(M.getId()).append(" : ")
-		  .append(M.getLastName()).append(", ").append(M.getFirstName()).append("\n")
-		  .append("  Book ").append(B.getBookId()).append(" : " )
-		  .append(B.getTitle()).append("\n")
-		  .append("  DueDate: ").append(sdf.format(D)).append("\n")
-		  .append("  State: ").append(state);		
-		return sb.toString();
-	}
-
-
-	public Member Member() {
-		return M;
-	}
-
-
-	public Book Book() {
-		return B;
-	}
-
-
-	public void DiScHaRgE() {
-		state = LOAN_STATE.DISCHARGED;		
-	}
+
+    public static enum LoanState { CURRENT, OVER_DUE, DISCHARGED };
+
+    private int loanId;
+    private Book bookId;
+    private Member memberId;
+    private Date dueDate;
+    private LoanState loanState;
+
+
+    public loan(int loanId, Book booksLoaned, Member member, Date loanDueDate) {
+        this.loanId = loanId;
+        this.bookId = booksLoaned;
+        this.memberId = member;
+        this.dueDate = loanDueDate;
+        this.loanState = LoanState.CURRENT;
+    }
+
+
+
+    public void checkOverDueLoan() {
+        if (this.loanState == LoanState.CURRENT &&
+            Calendar.getInstance().getDate().after(this.dueDate)) {
+            this.loanState = LoanState.OVER_DUE;
+        }
+    }
+
+
+    public boolean isOverDue() {
+        return this.loanState == LoanState.OVER_DUE;
+    }
+
+
+    public Integer getLoanId() {
+        return this.loanId;
+    }
+
+
+    public Date getDueDate() {
+        return this.dueDate;
+    }
+
+
+    public String toString() {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+
+        StringBuilder concatenate  = new StringBuilder();
+        concatenate .append("Loan:  ").append(this.loanId).append("\n")
+          .append("  Borrower ").append(this.memberId.getId()).append(" : ")
+          .append(this.memberId.getLastName()).append(", ").append(this.memberId.getFirstName()).append("\n")
+          .append("  Book ").append(this.bookId.getBookId()).append(" : " )
+          .append(this.bookId.getTitle()).append("\n")
+          .append("  DueDate: ").append(dateFormat.format(this.dueDate)).append("\n")
+          .append("  State: ").append(this.loanState);
+        return concatenate .toString();
+    }
+
+
+    public Member getMemberId() {

```

Comment:
DEFECT 7: Expanding on the earlier comment, this should be something more meaningful such as "getLoanMember" as it returns a Member instance and not specifically an ID.

Reply:
D7: Strange, it's already showing this:
public Member getLoanMember() {
        return this.loanMember;
    }